ktor-http/common/src/io/ktor/http/HttpHeaderValueParser.kt

```diff
@@ -155,36 +155,43 @@ private fun parseHeaderValueParameter(text: String, start: Int, parameters: Lazy
 
 
 private fun parseHeaderValueParameterValue(value: String, start: Int): Pair<Int, String> {
-    var pos = start
-    while (pos <= value.lastIndex) {
-        when (value[pos]) {
-            '"' -> return parseHeaderValueParameterValueQuoted(value, pos + 1)
-            ';', ',' -> return pos to value.subtrim(start, pos)
-            else -> pos++
+    if (value.length == start) {
+        return start to ""
+    }
+
+    var position = start
+    if (value[start] == '"') {
+        return parseHeaderValueParameterValueQuoted(value, position + 1)
```

Comment:
What would happen if `value == "\""`?

Reply:
I hope that case is being handled on the call side?

Reply:
I can't get the difference here. Could you clarify?

Reply:
According to the way `parseHeaderValueParameterValueQuoted` works, the result would be the same for `"\""` and `"\"\""` (in both cases result would be a mapping from size to an empty string, and the last parsed symbol is `\"` in both cases). 
I guess, `\"\"` must be a quote of empty string and `\"` must be either an error or a text that just contains single  quotation mark.
So I think result must be empty for `"\"\""` and `\"` for `\"`

Reply:
Thanks. I will add the test and fix the problem.